ONSdigital/respondent-home-ui
# Make URL path prefix configurable

# Motivation and Context

The application needs to handle an optional URL path component prefix which can be configured when deployed. The `/info` route is needed for health-checks, so do not prefix anything for this route. We were unable to configure the manifest to accept variable substitution with Concourse when the app is deployed, so we explicitly ignore this via the code.

The [pipeline changes](https://github.com/ONSdigital/ras-deploy/compare/add-url-path-prefix-for-rh?expand=1) are in place ready for this PR.

# What has changed



* Changed app creation behaviour to allow URL prefix when app routes are mapped.
* Ignore URL prefixing for `/info` for health-checks. 

# How to test?



1. Edit your `.env` file and set `URL_PATH_PREFIX=/test-prefix`
2. Open Respondent Home UI with the prefix, e.g. `localhost:9092/test-prefix`
3. Ensure the page renders correctly with GETs and POSTs.
4. Repeat above steps but remove `URL_PATH_PREFIX` and ensure page renders at root level with GETs and POSTs.

# Links



[Trello](https://trello.com/c/vteevzP4/219-sus012-ext-setup-url)
